test/migration_repos_selection_dialog_test.rb

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/env rspec
+
+require_relative "spec_helper"
+
+describe Registration::UI::MigrationReposSelectionDialog do
+  subject { Registration::UI::MigrationReposSelectionDialog }
+
+  describe ".run" do
+    before do
+      allow(Yast::Pkg).to receive(:GetUpgradeRepos).and_return([1, 2])
+      allow(Yast::Pkg).to receive(:SourceGetCurrent).and_return([0, 1, 2, 3])
+      allow(Yast::Pkg).to receive(:SourceGeneralData).and_return(
+        "name" => "name", "url" => "https://example.com")
+
+      # check the displayed content
+      expect(Yast::Wizard).to receive(:SetContents) do |_title, content, _help, _back, _next|
+        # do a simple check: convert the term to a String
+        # an unselected repository
+        expect(content.to_s).to include("item (`id (0), \"name (https://example.com)\", false)")
```

Comment:
I do not like this as it very depends on term serialization. There is stronger matchers for term: https://github.com/yast/yast-ruby-bindings/blob/master/src/ruby/yast/term.rb#L69 or anything from Enumberable.